reposerver/src/main/scala/com/advancedtelematic/tuf/reposerver/http/RepoResource.scala

```diff
@@ -114,9 +115,15 @@ class RepoResource(keyserverClient: KeyserverClient, namespaceValidation: Namesp
        .map(_ => repoId)
    }
 
+  private val malformedRequestContentRejectionHandler = RejectionHandler.newBuilder().handle {
+    case MalformedRequestContentRejection(msg, _) => complete((StatusCodes.BadRequest, msg))
+  }.result()
+
   private def createRepo(namespace: Namespace, repoId: RepoId): Route =
-    entity(as[CreateRepositoryRequest]) { keyType =>
-      createRepo(namespace, repoId, keyType.keyType)
+    handleRejections(malformedRequestContentRejectionHandler) {
```

Comment:
Can use `DefaultRejectionHandler`?

Reply:
Good idea, doesn't work out of the box since DefaultRejectionHandler doesn't handle MalformedRequestContentRejection(_, DecodingFailure()) which occurs here. I can extend DefaultRejectionHandler though.

Reply:
Doesn't work, either, since our DefaultRejectionHandler.rejectionHandler falls back on the Akka RejectionHandler.default which handles RequestEntityExpectedRejection which fails for an empty entity instead of going to the default case.